tests/test_time_series_graph.py

```diff
@@ -1865,3 +1865,77 @@ def test_get_nodes_for_variable_name(self):
             set(cg.get_nodes_for_variable_name('b')), set(cg.get_nodes(['b lag(n=1)', 'b', 'b future(n=1)']))
         )
         self.assertSetEqual(set(cg.get_nodes_for_variable_name('c')), set(cg.get_nodes(['c future(n=1)'])))
+
+    def test_node_cache_upon_construction(self):
+        input_nodes = ['A lag(n=2)', 'A lag(n=1)', 'B']
+
+        ts_graph = TimeSeriesCausalGraph(input_list=input_nodes, fully_connected=False)
+
+        nodes_in_lag_cache = []
+        for key, nodes in ts_graph.lag_to_nodes.items():
+            for node in nodes:
+                nodes_in_lag_cache.append(node.identifier)
+        self.assertListEqual(nodes_in_lag_cache, input_nodes)
+        nodes_in_variable_cache = []
+        for key, nodes in ts_graph.variable_name_to_nodes.items():
+            for node in nodes:
+                nodes_in_variable_cache.append(node.identifier)
+        self.assertListEqual(nodes_in_variable_cache, input_nodes)
```

Comment:
this is a pretty strange way to test these dicts I think. Why not test for dict equality? e.g. `self.assertDictEquals(ts_graph.lag_to_nodes, dict(0=[ts_graph['B']], 1=[ts_graph['A lag(n=1)']], 2=[ts_graph['A lag(n=2)'))` or something like that?

Because currently you don't actually test that nodes appear under correct lags.

Same goes for all the tests below.